packages/cli/src/commands/common/accounts.ts

```
@@ -29,6 +31,21 @@ export const ACCOUNTS_OPTION: KeyedOption<AccountsArg> = {
   type: 'stringsList',
 }
 
+export const getAdaptersTags = (adapters: string[]): Partial<Tags> => (
+  Object.fromEntries(adapters
+    .filter(adapter => getSupportedServiceAdapterNames().includes(adapter))
+    .map(adapter => [`adapter-${adapter}`, true]))
+)
+
+export const getTagsForAccounts = (
+  args: { workspace: Workspace } & AccountsArg & EnvArg
+): Partial<Tags> => {
+  const { workspace, accounts, env } = args
+  const envAccounts = workspace.accounts(env)
+  const validAccounts = _.isEmpty(accounts) ? envAccounts : _.intersection(accounts, envAccounts)
+  return getAdaptersTags(validAccounts.map(workspace.getServiceFromAccountName))
```

feels like this should use the same logic as `getAndValidateActiveAccounts`.
there are some non trivial choices here (e.g - when no accounts are specified it means "all accounts")

not sure how to share this logic while keeping the error cases below as they are (worst case, we could use a "try" block here and ignore the errors?)

---

The no accounts => all accounts is the semantic of ACCOUNTS_OPTION

I can use try with getAndValidateActiveAccounts, I thought it's very inelegant. (same with adding a shouldThrow argument)

---

and there is no way to extract the logic from `getAndValidateActiveAccounts` to make it reusable?
I didn't think about it too much, if you say there isn't a way, then I agree what you did is probably better than catching the exception

---

I changed `getAndValidateActiveAccounts` to only check for errors, and use `getValidAccounts`. to do the actual logic